src/compiler/crystal/crystal_path.cr

```diff
@@ -127,6 +140,8 @@ module Crystal
         else
           if filename.ends_with?(".cr")
             files << full_name
+          elsif filename.ends_with?(".ccr")
+            files << CCR.process(full_name)
```

Comment:
```cr
case ::Path[filename].extension
when ".cr"  then files << full_name
when ".ccr" then files << CCR.process(full_name)
end
```

Reply:
Not really an optimization, getting the extension needs to allocate a string, whereas `ends_with?` is just a super fast memcmp.

Also, I wouldn't worry right now about how clean or optimal the code is.